consensus/api/proto/consensus_admin.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+// Consensus administrative data types and service descriptors.
+// There are APIs related to managing a consensus-service instance and getting information out of it.
+// Some of them are sensitive, and as such these are not intended to be exposed to the world.
+
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+
+package consensus_admin;
+
+service ConsensusAdminAPI {
+    // Get Prometheus metrics.
+    rpc GetPrometheusMetrics (google.protobuf.Empty) returns (GetPrometheusMetricsResponse);
+
+    // Get information such as build info, logging configuration, etc.
+    rpc GetInfo (google.protobuf.Empty) returns (GetInfoResponse);
+
+    // Set RUST_LOG.
+    rpc SetRustLog (SetRustLogRequest) returns (google.protobuf.Empty);
+
+    // Logs a test error message.
+    rpc TestLogError (google.protobuf.Empty) returns (google.protobuf.Empty);
+}
+
+message GetPrometheusMetricsResponse {
+    // Prometheus-encoded metrics.
+    // https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md
+    string metrics = 1;
+}
+
+message GetInfoResponse {
+    // JSON-encoded build information.
+    string build_info_json = 1;
+
+    // JSON-encoded configuration (see `consensus_service::Config`).
+    string config_json = 2;
+
+    // JSON-encoded network configuration (see `consensus_service::NetworkConfig`).
+    string network_json = 3;
```

Comment:
Any reason not to `derive(Message, Serialize, Deserialize)` on those structures and include them here?

Reply:
I am not sure - I wanted to get this out of the way quickly. I expect deriving nested structures (quorum set stuff) is going to get tricky, as well as stuff such as https://github.com/mobilecoinofficial/mobilecoin/blob/master/consensus/service/src/config.rs#L38 is going to get messy.

I'd prefer to proceed as-is and try and improve at a later PR since I am not going to have time in the next few days to address this due to iOS work, and I'd really like this to not go stale (since it touches both repos' deployment scripts which are frustrating to test).

Reply:
Right, OK, gotcha.

Reply:
Thanks!